libs/api-client-framework/src/blockscout/config.rs

```diff
@@ -0,0 +1,111 @@
+use reqwest::header::HeaderValue;
+use reqwest_middleware::Middleware;
+use reqwest_retry::{policies::ExponentialBackoff, RetryTransientMiddleware};
+use serde::{Deserialize, Deserializer};
+use std::{fmt, fmt::Formatter, sync::Arc, time::Duration};
+
+#[derive(Clone, Deserialize)]
+pub struct Config {
+    pub url: url::Url,
+    #[serde(default, deserialize_with = "deserialize_api_key")]
+    pub api_key: Option<HeaderValue>,
+    /// The maximum time limit for an API request. If a request takes longer than this, it will be
+    /// cancelled. Defaults to 30 seconds.
+    #[serde(default = "defaults::http_timeout")]
+    pub http_timeout: Duration,
```

Comment:
How timeout is deserialized in that case (as seconds)? Also, can we make the default value configurable?